.checkstyle/import-control.xml

```diff
@@ -78,6 +78,7 @@ We also control imports only in production classes and not in tests. This is con
         <allow class="io.strimzi.operator.common.Annotations" />
         <allow class="io.strimzi.operator.cluster.ClusterOperatorConfig" />
         <allow class="io.strimzi.operator.cluster.PlatformFeaturesAvailability" />
+        <allow class="io.strimzi.operator.cluster.operator.resource.KafkaMetadataStateManager" />
```

Comment:
Looking at the code while working on something else, it looks like the majority of classes like this are in this package. So maybe you can ignore this comment for the time being. I still think the placement makes no sense, but we should probably deal with it _globally_ and not just with this class 🤷.

Reply:
Since the initial development the class was in the `assembly` package as you were suggesting. When I needed to use the `KafkaAgentClient` in it, I noticed that such a class is package-private. Tbh I was not sure to make the `KafkaAgentClient` public so I did the other way around, moving the state manager under `resource`.

Reply:
Well, as I said, I do not think it makes any sense for the KafkaAgentCLient to be in the `resource` package. It is not managing any resource. But as suggested, I think you should proceed as it is and we can move all of them later 🤷